torch_xla/csrc/lowering_context.cpp

```diff
@@ -154,9 +154,19 @@ void LoweringContext::SetResult(size_t index, xla::XlaOp op) {
 
 xla::StatusOr<xla::XlaComputation> LoweringContext::BuildXla() {
   xla::StatusOr<xla::XlaComputation> xla;
-  if (!root_tuple_.empty()) {
+  if (!root_tuple_.empty() & (root_tuple_.size() > 1)) {
     xla::XlaOp root = xla::Tuple(builder(), root_tuple_);
     xla = builder()->Build(root);
+  } else if (!root_tuple_.empty() & (root_tuple_.size() == 1)) {
```

Comment:
I think we should condition on `get_name_string()`. Add this check at the top and build for while loop `if get_name_string() == "condctx" or get_name_string() == "bodyctx"; otherwise, you can keep the original build logic.

Have your logic for while loop build in a separate private method, and call it if ``if get_name_string() == "condctx" or get_name_string() == "bodyctx"` is true. 

So you can keep `BuildXla()` simple.

Reply:
Thanks, make sense and updated in the newest commit, due to the `logic for while loop` is one simple line code, we run it directly without warping it in a separate private method